sdk/ml/azure-ai-ml/setup.py

```diff
@@ -69,7 +69,7 @@
         "msrest>=0.6.18",
         "azure-core<2.0.0,>=1.23.0",
         "azure-mgmt-core<2.0.0,>=1.3.0",
-        "marshmallow<4.0.0,>=3.5",
+        "marshmallow<3.20.0,>=3.5",
```

Comment:
We do not pin dependencies to minor versions. Compatibility must be maintained will all versions <4.0.

Reply:
Having read the context in more detail - this is definitely a problem.
Fortunately we only have a few months before we probably drop 3.7 too.
If the tests are currently failing due to a re-record, that should probably be resolved separately.

Reply:
@mccoyp to temporarily workaround this and get this CI green, I recommend adding `marshmellow<3.20` to their `dev_requirements.txt`. This should get CI green while nightly will still be red.

`latest dependency` will still melt unless we add an override, but I think you should probably just let that one fail until they resolve it.

Reply:
Got it, that makes sense. We can also probably get the affected tests passing with both the latest and pinned versions by disabling body matching; I can open an issue to track removing that body matching once 3.7 is dropped